src/Contexts/ProductListContext/GymAccessories.js

```diff
@@ -0,0 +1,152 @@
+const { createContext, useContext, useReducer } = require("react");
+// created context for home scrren products
+export const gymAccessoriesContext = createContext();
+// initial state
+const initialStateOfGymAccessories = {
+  initialGymAccessories: [],
+  loading:true,
+  filterItems: {
+   sort:"",
+  stock:"",
+    rating: false,
+ delivery:"",
+    productTags: [],
+    price_range:null
+  }
+};
+
+// usereducer finctions
+const GymAccessoriesReduceFun = (state, action) => {
```

Comment:
here filter and sort function is repeating so putting in one file and access it from there will reduce your work and your code look nice

Reply:
Hey Shivam
Thank you for your review,
I really wanted to do that, i have individual filter since i dont want to loose filtered state of one category if user apply filter on other category/pages since all filtered state will be depend on one function so i wanted to persist that  state and storing it in localstorage would make that heavy so,
Thank you
Akshay